### PR TITLE
Fix for #Pydev 1090: MyPy integration: "#noqa" should silence error

### PR DESCRIPTION
**Pull request by oluiscabral**

### CODE REVIEW

File: plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java [hunk 1]
```diff
@@ -294,6 +300,19 @@ public void afterRunProcess(String output, String errors, IExternalCodeAnalysisS
                     }
                     if (markerSeverity != -1) {
                         line = Integer.parseInt(outputLine.substring(m.start(2), m.end(2))) - 1;
+                        String lineContents = PySelection.getLine(document, line);
```
Comment (fabioz) [inline plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java:303]: I know this works, but I think we can be a bit less strict about it in this case.

You can just use: `org.python.pydev.core.CheckAnalysisErrors.isCodeAnalysisErrorHandled(lineContents, "type")` for this.

**Review by fabioz — CHANGES_REQUESTED** [review]

File: plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java [hunk 2]
```diff
@@ -342,7 +331,9 @@ public void afterRunProcess(String output, String errors, IExternalCodeAnalysisS
                 Log.log(e);
             }
         }
-        for (MessageInfo messageInfo : lineColToMessage.values()) {
+        for (
```
Comment (fabioz) [inline plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java]: Please revert the strange indent...

**Review by fabioz — CHANGES_REQUESTED** [review]
Just a really minor nitpick ;)

File: plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java (hunk 2 above)
Comment (oluiscabral) [reply plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java]: Yup, I did it in https://github.com/fabioz/Pydev/pull/294/commits/2d92da641b74a6cb41665be162aed7caf1f490ce

File: plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java [hunk 3]
```diff
@@ -236,6 +239,7 @@ void createMypyProcess(IExternalCodeAnalysisStream out)
 
     @Override
     public void afterRunProcess(String output, String errors, IExternalCodeAnalysisStream out) {
+        ParsingUtils parsingUtils = ParsingUtils.create(document);
```
Comment (fabioz) [inline plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java:242]: This is no longer needed.

**Review by fabioz — CHANGES_REQUESTED** [review]

File: plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java (hunk 3 above)
Comment (oluiscabral) [reply plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java]: That's right, I forgot it

Comment (oluiscabral) [reply plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java]: I removed it in the last commit ;)

**Pull request merged**